Src/VimCore/CommandUtil.fs

```diff
@@ -888,7 +888,8 @@ type internal CommandUtil
     /// Used for the several commands which make an edit here and need the edit to be linked
     /// with the next insert mode change.
     member x.CreateTransactionForLinkedChange name action =
-        let transaction = _undoRedoOperations.CreateLinkedUndoTransaction name
+        let flags = LinkedUndoTransactionFlags.EndsWithInsert
+        let transaction = _undoRedoOperations.CreateLinkedUndoTransactionWithFlags name flags

```

Comment:
Why do we want to do this for all the command based commands? I'm trying to think of one that would be affected by this.

Reply:
Perhaps the function name isn't specific enough right now, but `CreateTransactionForLinkedChange` is only called from three places, and all of them are defined to end in insert mode.

Reply:
I'm seeing now the comment above `CreateTransactionForLinkedChange` that says:

> and need the edit to be linked with the next insert mode change

which is exactly what the flag specifies.